Library/Homebrew/dev-cmd/livecheck.rb

```diff
@@ -51,7 +51,7 @@ def livecheck
     formulae_to_check = if args.tap
       Tap.fetch(args.tap).formula_names.map { |name| Formula[name] }
     elsif args.installed?
-      Formula.installed
+      Formula.installed.map { |formula| Formula[formula.full_name] }
```

Comment:
I think `Formula.installed` is correct in behaving the way it does. It is supposed to return what is installed – and in the example I gave, I had a `head` version installed, which is what it returns.

It's just that `livecheck` isn't a 'local' update checker, but a developer command probably used by contributors who want to bump formula versions, for example in `homebrew/core`.

The scenario this PR fixes is an edge case, where the person has the `head`-version installed (maybe to get rid of a bug fix in the latest stable version), and wants to check for the next stable version to be released with `livecheck`.

Reply:
Ok. I think at least this warrants a comment but it's not clear to the casual reader why this wouldn't return the same as `Formula.installed`. An optional argument for it may make more sense in making this more explicit?

Reply:
I can add a comment explaining the situation, yes. As for the argument, I'm not too sure what it would be, but I don't think there's a strong case for one. Livecheck identifies the latest upstream versions of formulae, so `--installed` just means "fetch the latest versions of formulae that I have on my system."

I don't feel too strongly about not having the option though, so I will add it if it makes more sense.